src/workers/gh_client.coffee

```diff
@@ -0,0 +1,162 @@
+
+  teams:
+    create: (gh_teams_opts) ->
+      return Promise.all(gh_teams_opts.map((gh_team_opts) ->c.team.create(gh_team_opts)))
+    repo:
+      add: (gh_team_ids, gh_repo_fullname) ->
+        return Promise.all(gh_team_ids.map((gh_team_id) -> c.team.repo.add(gh_team_id, gh_repo_fullname)))
+      remove: (gh_team_ids, gh_repo_fullname) ->
+        return Promise.all(gh_team_ids.map((gh_team_id) -> c.team.repo.remove(gh_team_id, gh_repo_fullname)))
+    user:
+      add: (gh_team_ids, gh_username) ->
+        return Promise.all(gh_team_ids.map((gh_team_id) -> c.team.user.add(gh_team_id, gh_username)))
+      remove: (gh_team_ids, gh_username) ->
+        return Promise.all(gh_team_ids.map((gh_team_id) -> c.team.user.remove(gh_team_id, gh_username)))
+
+  repo:
+    create: (gh_repo_opts) ->
+      ###
+      gh_repo_opts is a hash that at minimum must contain a `name` key with the repo's name.
+      Can include any other opts specified at https://developer.github.com/v3/repos/#create
+      ###
+      gh_repo_opts = _.defaults(gh_repo_opts, {
+        description: "",
+        homepage: "https://github.com/" + gh_conf.ORG_NAME,
+        private: false,
+        has_issues: true,
+        has_wiki: true,
+        has_downloads: true
+      })
+      url = '/organizations/' + gh_conf.ORG_ID + '/repos'
+      return git.post({
+        url: url,
+        json: gh_repo_opts,
+        body_only: true,
+      }).catch((err) ->
+        if err.msg?.errors?[0]?.message == 'name already exists on this account'
+          return git.find_one(url, (repo) -> return repo.name == gh_repo_opts['name'])
+        else
+          return Promise.reject(err)
+      )
+    pushTemplate: (repo_data) ->
+      # if this is not a brand new repo, then we won't push to it.
+      if repo_data.created_at != repo_data.updated_at or repo_data.updated_at != repo_data.pushed_at
+        return Promise.resolve(repo_data)
+      return exec('git init', {cwd: TEMPLATE_DIR}).catch((err) ->
+        if err.code == 'ENOENT'
+          return mkdir(TEMPLATE_DIR).then(() ->
+            exec('git init', {cwd: TEMPLATE_DIR})
+          )
+        else
+          return Promise.reject(err)
+      ).then(() ->
+        exec('git pull "' + gh_conf.TEMPLATE_REPO + '"', {cwd: TEMPLATE_DIR})
+      ).then(() ->
+        push_repo_url = get_authenticated_repo_url(repo_data.clone_url)
+        exec('git push "' + push_repo_url + '" master', {cwd: TEMPLATE_DIR})
+      ).catch((err) ->
+        console.log('we had an error')
+        pushTemplate(repo_data)
+      ).then(() ->
+        Promise.resolve(repo_data)
+      )
+
+    createPush: (gh_repo_opts) ->
+      # create the repo, if it doesn't exist, then push the template
+      # if new repo
+      c.repo.create(gh_repo_opts).next((repo_data) ->
+        return git.repo.pushTemplate(repo_data)
+      )
+
+  client: git
+
+# nodeifyAll = (nested_method_hash) ->
```

Comment:
Ah. this should actually be removed. I misunderstood how, exactly, nodeify works.